### PR TITLE
717 component edit add confirm window

### LINKED ISSUE #717: Add confirmation window when exiting add component window
In case a user accidentally closes the window half-way through adding/editing a component

### PR DESCRIPTION
**Pull request by kmurica**
Closes #717

### CODE REVIEW

**Comment by kmurica** [comment]
Some UI tests are failing due to my update. Have to check it tomorrow. No need to start the review until I resolve the test issues.

**Comment by kmurica** [comment]
@matthew-d-jones Jenkins is green now for this PR. Ready for review.

File: nexus_constructor/main_window.py [hunk 1]
```diff
@@ -198,3 +198,41 @@ def show_add_component_window(self, component: Component = None):
...
+    a message box in the component editor to the users,
+    asking if they are sure to quit editing component when exiting.
+    """
+
+    def __init__(self):
+        super().__init__()
+        self._is_accepting_component = True
+
+    def enable_msg_box(self):
+        self._is_accepting_component = True
```
Comment (umithardal) [inline nexus_constructor/main_window.py:215]: This is also done in the constructor. Do we need them both?

Comment (kmurica) [reply nexus_constructor/main_window.py:215]: Yes, you are correct. I was planning to remove it but forgot. :) As it is only relevant when closing the window, setting to true again makes no sense. I will remove it.

**Review by umithardal — APPROVED** [review]

**Pull request merged**